context-specific ablation study (#38)
thesis20/thesis_paper: sections/experiments/experiments-chapter.tex

\resizebox{\textwidth}{!}{%
\begin{tabular}{c|l|cccc|cc}
Dataset & Metric & FM & \multicolumn{1}{c}{NFM} & \multicolumn{1}{c}{CFM} & \multicolumn{1}{c|}{CSGCN-IS} & \multicolumn{1}{c}{CSGCN-ADJ} & \multicolumn{1}{c|}{Impr.} \\ \hline

> \multirow{4}{*}{YelpNC} & HR@20  & 0.0075  & 0.0142  & 0.0070 & 0.0625 & 0.0761  &  \\

Author: ```suggestion
\multirow{4}{*}{Yelp-NC} & HR@20  & 0.0075  & 0.0142  & 0.0070 & 0.0625 & 0.0761  &  \\
```

        };

        \addplot plot coordinates {
            (20,0.04266)
            (40,0.04749)
            (60,0.04661)
            (80,0.04925)
            (100,0.05145)
            (120,0.04925)
            (140,0.05277)
            (160,0.05453)
            (180,0.05673)
            (200,0.05717)
        };

        \addplot plot coordinates {
            (20,0.04969)
            (40,0.04705)
            (60,0.05145)
            (80,0.05453)
            (100,0.05585)
            (120,0.05453)
            (140,0.05893)
            (160,0.06069)
            (180,0.06157)
            (200,0.06157)
        };

        \addplot plot coordinates {
            (20,0.04749)
            (40,0.04661)
            (60,0.04749)
            (80,0.05101)
            (100,0.05013)
            (120,0.05057)
            (140,0.05365)
            (160,0.05145)
            (180,0.05277)
            (200,0.05497)
        };
        \legend{Sideinfo + context,Sideinfo,Context,Nothing}
        \end{axis}
    \end{tikzpicture}
    \caption{The performance on HR@20 of CSGCN-IS on the Yelp-NC dataset with different types of input.}
    \label{fig:ablation_study_context_specific_2}
\end{figure}

> \Cref{fig:ablation_study_context_specific_1} shows the HR@20 for CSGCN-ADJ on YelpNC with different input. 
> We see that when side-information is included in the input for CSGCN-ADJ it does not change the performance much.
> Another interesting thing to note is how the model performs worse when context is used in the input, even though we are doing context-specific recommendations.
> This could be because the CSGCN-ADJ model does not actually model context in way that fits the definition of context defined in \Cref{subsec:define_context_sideinfo}.
> In CSGCN-ADJ, context can instead be viewed as a type of side-information, since it is modelled in a somewhat similar way.
> \\\\
> \Cref{fig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on YelpNC with different input.
> For CSGCN-IS we see a slight increase in performance when side-information is included, compared to not having side-information in the input.
> This shows that CSGCN-IS is able to use the side-information to better connect user and item nodes, increasing the collaborative signal.
> We also see that for this model, including context in the input does make a significant different for the performance in a context-specific setting.
> This is because context is modelled with finer granularity than in CSGCN-ADJ, meaning that each item and context value combination is represented by a vector.
> The result of this, is that context can have different effect on the various items, resulting in more accurate context-specific recommendations.

Reviewer: ```suggestion
\Cref{fig:ablation_study_context_specific_1} shows the HR@20 for CSGCN-ADJ on YelpNC with different input. 
We see that when side-information is included in the input for CSGCN-ADJ it does not change the performance much.
Another interesting thing to note is how the model performs worse when the context is used in the input, even though we are doing context-specific recommendations.
This could be because the CSGCN-ADJ model does not model context in a way that fits the definition of context defined in \Cref{subsec:define_context_sideinfo}.
In CSGCN-ADJ, context can instead be viewed as a type of side-information, since it is modeled in a somewhat similar way.
\\\\
\Cref{fig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on YelpNC with different input.
For CSGCN-IS we see a slight increase in performance when side-information is included, compared to not having side-information in the input.
This shows that CSGCN-IS is able to use the side-information to better connect user and item nodes, increasing the collaborative signal.
We also see that for this model, including the context in the input does make a significant difference for the performance in a context-specific setting.
This is because context is modeled with finer granularity than in CSGCN-ADJ, meaning that each item and context value combination is represented by a vector.
The result of this is that context can have a different effect on the various items, resulting in more accurate context-specific recommendations.
```

Side-information, however, performs the best on this metric on this dataset, which suggests that side-information in CSGCN-ADJ is able to slightly improve results compared to having no extra input.
\\
On \Cref{fig:ablation_study_non_context_2} the NDGC@20 results of ML1M are shown for CSGCN-ADJ with different types of input. 
For this dataset, the model that uses neither the context nor side-information for the input is actually the best performing version.
Side-information and context are not able to improve the performance of CSGCN-ADJ on ML1M in a non-context-specific setting.
Since both \Cref{fig:ablation_study_non_context_1,fig:ablation_study_non_context_2} show that the model is worse with context included, it can be concluded that context does not increase the performance of CSGCN-ADJ in a non-context specific setting.
Side-information, however, is able to increase performance of the model in a non-context specific setting, but for this dataset the side-information does not provide any additional collaborative signal.

\subsection{Performance Comparison (RQ3)}

> To answer whether side-information improves the results for context-aware recommendations, we performed an ablation study on the YelpNC dataset where we ran the CSGCN models with and without side-information to test the effect it has on the performance.

Reviewer: ```suggestion
To answer whether side-information improves the results for context-aware recommendations, we performed an ablation study on the YelpNC dataset.
In the study, the CSGCN models were run with and without side-information to test the effect it has on the performance.
```

> This was also done for context input to test how it affects the performance in a context-specific setting.

Reviewer: ```suggestion
Additionally, the experiment was conducted on the context input to examine how much it affects the performance in a context-specific setting.
```

            (80,0.04925)
            (100,0.05145)
            (120,0.04925)
            (140,0.05277)
            (160,0.05453)
            (180,0.05673)
            (200,0.05717)
        };

        \addplot plot coordinates {
            (20,0.04969)
            (40,0.04705)
            (60,0.05145)
            (80,0.05453)
            (100,0.05585)
            (120,0.05453)
            (140,0.05893)
            (160,0.06069)
            (180,0.06157)
            (200,0.06157)
        };

        \addplot plot coordinates {
            (20,0.04749)
            (40,0.04661)
            (60,0.04749)
            (80,0.05101)
            (100,0.05013)
            (120,0.05057)
            (140,0.05365)
            (160,0.05145)
            (180,0.05277)
            (200,0.05497)
        };
        \legend{Sideinfo + context,Sideinfo,Context,Nothing}
        \end{axis}
    \end{tikzpicture}
    \caption{The performance on HR@20 of CSGCN-IS on the Yelp-NC dataset with different types of input.}
    \label{subfig:ablation_study_context_specific_2}
    \end{subfigure}
    \caption{Ablation study for the input of CSGCN-ADJ and CSGCN-IS on Yelp-NC.}
\end{figure*}

> \Cref{subfig:ablation_study_context_specific_1} shows the HR@20 for CSGCN-ADJ on YelpNC with different input. 

Reviewer: ```suggestion
\Cref{subfig:ablation_study_context_specific_1} shows the HR@20 for CSGCN-ADJ on YelpNC with different combinations of input. 
```

> We see that when side-information is included in the input for CSGCN-ADJ it does not change the performance much.

Reviewer: ```suggestion
We see that when side-information is included in the input for CSGCN-ADJ it does not change the performance much compared to excluding it.
```

Another interesting thing to note is how the model performs worse when context is used in the input, even though we are doing context-specific recommendations.
This could be because the CSGCN-ADJ model does not actually model context in way that fits the definition of context defined in \Cref{subsec:define_context_sideinfo}.
In CSGCN-ADJ, context can instead be viewed as a type of side-information, since it is modelled in a somewhat similar way.
\\\\

> \Cref{subfig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on YelpNC with different input.

Reviewer: ```suggestion
\Cref{subfig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on YelpNC with different inputs.
```

> For CSGCN-IS we see a slight increase in performance when side-information is included, compared to not having side-information in the input.

Reviewer: ```suggestion
For CSGCN-IS, there is a slight increase in performance when side-information is included, compared to excluding it.
```

Side-information, however, performs the best on this metric on this dataset, which suggests that side-information in CSGCN-ADJ is able to slightly improve results compared to having no extra input.
\\
On \Cref{fig:ablation_study_non_context_2} the NDGC@20 results of ML1M are shown for CSGCN-ADJ with different types of input. 
For this dataset, the model that uses neither the context nor side-information for the input is actually the best performing version.
Side-information and context are not able to improve the performance of CSGCN-ADJ on ML1M in a non-context-specific setting.
Since both \Cref{fig:ablation_study_non_context_1,fig:ablation_study_non_context_2} show that the model is worse with context included, it can be concluded that context does not increase the performance of CSGCN-ADJ in a non-context specific setting.
Side-information, however, is able to increase performance of the model in a non-context specific setting, but for this dataset the side-information does not provide any additional collaborative signal.

\subsection{Performance Comparison (RQ3)}

> To answer whether side-information improves the results for context-aware recommendations, we performed an ablation study on the YelpNC dataset.

Reviewer 2: ```suggestion
To answer whether side-information improves the results for context-aware recommendations, we performed an ablation study on the Yelp-NC dataset.
```

            (60,0.04661)
            (80,0.04925)
            (100,0.05145)
            (120,0.04925)
            (140,0.05277)
            (160,0.05453)
            (180,0.05673)
            (200,0.05717)
        };

        \addplot plot coordinates {
            (20,0.04969)
            (40,0.04705)
            (60,0.05145)
            (80,0.05453)
            (100,0.05585)
            (120,0.05453)
            (140,0.05893)
            (160,0.06069)
            (180,0.06157)
            (200,0.06157)
        };

        \addplot plot coordinates {
            (20,0.04749)
            (40,0.04661)
            (60,0.04749)
            (80,0.05101)
            (100,0.05013)
            (120,0.05057)
            (140,0.05365)
            (160,0.05145)
            (180,0.05277)
            (200,0.05497)
        };
        \legend{Sideinfo + context,Sideinfo,Context,Nothing}
        \end{axis}
    \end{tikzpicture}
    \caption{The performance on HR@20 of CSGCN-IS on the Yelp-NC dataset with different types of input.}
    \label{subfig:ablation_study_context_specific_2}
    \end{subfigure}
    \caption{Ablation study for the input of CSGCN-ADJ and CSGCN-IS on Yelp-NC.}
\end{figure*}

> 

Reviewer 2: ```suggestion
```

> \Cref{subfig:ablation_study_context_specific_1} shows the HR@20 for CSGCN-ADJ on YelpNC with different combinations of input. 

Reviewer 2: ```suggestion
\Cref{subfig:ablation_study_context_specific_1} shows the HR@20 for CSGCN-ADJ on Yelp-NC with different combinations of input. 
```

> We see that when side-information is included in the input for CSGCN-ADJ it does not change the performance much compared to excluding it.

Reviewer 2: ```suggestion
The results show that when side-information is included in the input for CSGCN-ADJ it does not change the performance much compared to excluding it.
```

> Another interesting thing to note is how the model performs worse when context is used in the input, even though we are doing context-specific recommendations.

Reviewer 2: ```suggestion
Another interesting thing to note is how the model performs worse when context is included in the input, even though it is a context-specific setting.
```

> This could be because the CSGCN-ADJ model does not actually model context in way that fits the definition of context defined in \Cref{subsec:define_context_sideinfo}.

Reviewer 2: ```suggestion
This could be because the CSGCN-ADJ model does not actually model context in a way that fits the definition of context defined in \Cref{subsec:define_context_sideinfo}.
```

> In CSGCN-ADJ, context can instead be viewed as a type of side-information, since it is modelled in a somewhat similar way.

Reviewer 2: hvad mener vi med somewhat similar? måske beskriv det lidt mere præcist

Author: Der er hermed tilføjet en linje mere, ved ikke om det giver mere mening nu?

\\\\

> \Cref{subfig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on YelpNC with different inputs.

Reviewer 2: ```suggestion
\Cref{subfig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on Yelp-NC with different inputs.
```

For CSGCN-IS, there is a slight increase in performance when side-information is included, compared to excluding it.
This shows that CSGCN-IS is able to use the side-information to better connect user and item nodes, increasing the collaborative signal.
We also see that for this model, including context in the input does make a significant different for the performance in a context-specific setting.

> This is because context is modelled with finer granularity than in CSGCN-ADJ, meaning that each item and context value combination is represented by a vector.

Reviewer 2: skal vi kalde det vector eller embedding?

Author: Det er vel det samme :D men tror vi bruger lidt begge rundt omkring i rapporten

Reviewer 2: måske en trello med unify? 🤔

Author: done

> The result of this finer granularity, is that context can have different effect on the various items, resulting in more accurate context-specific recommendations.

Reviewer 2: ```suggestion
The result of this finer granularity is that context can have different effects on the various items, resulting in more accurate context-specific recommendations.
```

> This is because context is modelled with finer granularity than in CSGCN-ADJ, meaning that each item and context value combination is represented by a vector.

Author: ```suggestion
This is because context is modelled with finer granularity than in CSGCN-ADJ, meaning that each item and context value combination has an embedding.
```

> In general, we see that side-information can have a slight effect on the results, depending the type of model.

Reviewer 2: ```suggestion
In general, side-information can have a slight effect on the results depending on the type of model.
```

> However, the context seems to have a lot larger effect on the performance of the models, as would be expected for context-aware recommendations.

Reviewer 2: ```suggestion
However, the context seems to have a lot larger effect on the performance of the models, as would be expected for context-aware recommendation.
```

> Additionally, it may be worth noting that the context used for these experiments is mainly based on discretized values based on timestamps, which may or may not be a useful contextual dimension.

Reviewer 2: samme som før. måske en sætning om hvorfor de kunne være skrald

Author: Dunno det var @Moesgaarda der tilføjede sætningen så jeg fralægger mig alt ansvar

Reviewer: det er ikke min pr

Reviewer 2: hvad fanden

Author: jeg stemmer for @Eduardsen ordner det. synes du ikke også det @Moesgaarda??

\\\\
\Cref{subfig:ablation_study_context_specific_2} shows HR@20 for CSGCN-IS on Yelp-NC with different inputs.
For CSGCN-IS, there is a slight increase in performance when side-information is included, compared to excluding it.
This shows that CSGCN-IS is able to use the side-information to better connect user and item nodes, increasing the collaborative signal.
We also see that for this model, including context in the input does make a significant different for the performance in a context-specific setting.
This is because context is modelled with finer granularity than in CSGCN-ADJ, meaning that each item and context value combination has an embedding.
The result of this finer granularity is that context can have different effects on the various items, resulting in more accurate context-specific recommendations.
In general, side-information can have a slight effect on the results depending on the type of model.
However, the context seems to have a lot larger effect on the performance of the models, as would be expected for context-aware recommendation.

> Additionally, it may be worth noting that the context used for these experiments is mainly based on discretized values based on timestamps, which may or may not be a useful contextual dimension.

Reviewer 2: ```suggestion
Additionally, it may be worth noting that the context used for these experiments is mainly based on discretized values based on timestamps, which might not be a useful contextual dimension due to timezone differences.
```
